flytekit/remote/remote.py

```diff
@@ -802,9 +807,18 @@ def register_task(
                 domain=self.default_domain,
             )
 
-        ident = asyncio.run(
-            self._serialize_and_register(entity=entity, settings=serialization_settings, version=version)
-        )
+        try:
+            import nest_asyncio
```

Comment:
I would even go back to the non-async version of `_serialize_and_register` just so that it is harder to break jupyter support.